### PR TITLE
Updating Aggregator to optionally use transient local qos

### PR DESCRIPTION
**Pull request by vinnnyr**
# summary 
Optionally allow the diagnostics aggregator to subscribe via transient local qos

# test instructions
1. colcon build, source install, etc
2. `ros2 launch diagnostic_aggregator example.launch.py publish_statefully:=True`
3. Use `ros2 topic info /diagnostics_stateful --verbose` to confirm appropriate qos is being used.

### CODE REVIEW

**Review by Aposhian — APPROVED** [review]
This looks great!

File: diagnostic_aggregator/example/example_pub.py [hunk 1]
```diff
@@ -53,10 +53,15 @@ class DiagnosticTalker(Node):
...
         super().__init__('diagnostic_talker')
+        self.declare_parameter('publish_transient_local', False)
+        use_transient_local = self.get_parameter('publish_transient_local').get_parameter_value().bool_value
+        qos = qos_profile_system_default
+        if use_transient_local:
+            qos.durability = QoSDurabilityPolicy.TRANSIENT_LOCAL
         self.i = 0
         self.pub = self.create_publisher(DiagnosticArray,
                                          '/diagnostics',
-                                         qos_profile_system_default)
```
Comment (adamsj-ros) [inline diagnostic_aggregator/example/example_pub.py:59]: I think you need two topics to subscribe -- one supporting legacy (backwards compatible) on `/diagnostics` and another supporting stateful on `/diagnostics_ros2`. This is especially important until the diagnostic updater is also updated to support these QoS policies.

Comment (vinnnyr) [reply diagnostic_aggregator/example/example_pub.py:59]: yeah, the way it currently is (backwards compatibility handled by solely a parameter) is not going to work for the use case. so I will do the two topic way.

**Review by Aposhian — APPROVED** [review]
This looks good.

**Review by adamsj-ros — APPROVED** [review]
Looks good.

**Pull request merged**